agent.ask() not retrieving value

## Checklist

- [V ] I have included information about relevant versions
- [V ] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce

**Calling agent.ask() outside of faust from an async function.**

I firstly run the faust app via `faust -A simplefaust worker -l info`

Then I have the following async function:

```python3
async def s_faust():
    from simplefaust import balance_agent
    print('getting balance')
    balance = await balance_agent.ask(
        DummyOrder(order_id='AB001', amount=0)
    )
    print(f'balance is {balance}')


if __name__=='__main__':
      loop = asyncio.get_event_loop()
      loop.run_until_complete(s_faust())

```




## Expected behavior

agent.ask() is supposed to retrieve value from the faust agent but it never retrieve any

## Actual behavior

faust app successfully executed the stream process but it never return to the caller

## Full traceback

```
Task exception was never retrieved
future: <Task finished coro=<Service._execute_task() done, defined at /opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py:800> exception=RuntimeError('Non-thread-safe operation invoked on an event loop other than the current one')>
Traceback (most recent call last):
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 802, in _execute_task
    await task
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/faust/transport/consumer.py", line 176, in _fetcher
    await consumer._drain_messages(self)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/faust/transport/consumer.py", line 1172, in _drain_messages
    callback(message), self.suspend_flow.wait()
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 715, in wait_first
    f.result()  # propagate exceptions
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/faust/transport/conductor.py", line 274, in on_message
    return await get_callback_for_tp(message.tp)(message)
KeyError: TopicPartition(topic='simplefaust_topic', partition=0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 813, in _execute_task
    await self.crash(exc)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 848, in crash
    child._crash(reason)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 855, in _crash
    node._crash(reason)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 855, in _crash
    node._crash(reason)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/services.py", line 852, in _crash
    self._crashed.set()
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/site-packages/mode/utils/locks.py", line 53, in set
    fut.set_result(True)
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/asyncio/base_events.py", line 693, in call_soon
    self._check_thread()
  File "/opt/anaconda3/envs/smarter_ddd/lib/python3.7/asyncio/base_events.py", line 731, in _check_thread
    "Non-thread-safe operation invoked on an event loop other "
RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one

```

# Versions

* Python version 3.8.13
* Faust version faust-streaming 0.8.4
* Operating system MacOS 12.3
* Kafka version Confluent Cloud Kafka
* RocksDB version (if applicable) N/A

I am also seeing this error, and but it even occurs inside of an `@app.command`.

### Full traceback
```
[2022-09-15 16:31:37,313] [15503] [ERROR] [^-Consumer]: Drain messages raised: KeyError(TopicPartition(topic='f-reply-0448b5f2-9e68-42e1-9296-5eecb63433d5', partition=0))
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/faust/transport/consumer.py", line 1197, in _drain_messages
    await self.wait_first(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/mode/services.py", line 751, in wait_first
    f.result()  # propagate exceptions
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/faust/transport/conductor.py", line 274, in on_message
    return await get_callback_for_tp(message.tp)(message)
KeyError: TopicPartition(topic='f-reply-0448b5f2-9e68-42e1-9296-5eecb63433d5', partition=0)
[2022-09-15 16:31:37,320] [15503] [ERROR] [^-Fetcher]: Crashed reason=KeyError(TopicPartition(topic='f-reply-0448b5f2-9e68-42e1-9296-5eecb63433d5', partition=0))
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/mode/services.py", line 835, in _execute_task
    await task
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/faust/transport/consumer.py", line 177, in _fetcher
    await consumer._drain_messages(self)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/faust/transport/consumer.py", line 1197, in _drain_messages
    await self.wait_first(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/mode/services.py", line 751, in wait_first
    f.result()  # propagate exceptions
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/cldap/lib/python3.8/site-packages/faust/transport/conductor.py", line 274, in on_message
    return await get_callback_for_tp(message.tp)(message)
KeyError: TopicPartition(topic='f-reply-0448b5f2-9e68-42e1-9296-5eecb63433d5', partition=0)
```

@wbarnha it is quite similar and I've reproduced the bug even with examples found in the documentation.

### Code
```python
from faust import App

app = App('example-app', broker='kafka://localhost:9092')

@app.agent()
async def example_agent(stream):
    async for event in stream:
        print('agent:', event)

from faust.cli import argument

@app.command(argument('event'))
async def ask_example(self, event):
    resp = await example_agent.ask(event)
    print('got:', resp)
```
### Ran
```
$ faust -A app.py ask-example "this is a test"
[2022-10-04 12:59:29,717] [37191] [ERROR] [^Worker]: Error: InvalidReplicationFactorError('Cannot create topic: f-reply-5d2dda89-c3d8-4831-b5b6-eb4ba7588d25 (38): Replication factor must be larger than 0.')
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/worker.py", line 279, in execute_from_commandline
    self.loop.run_until_complete(self._starting_fut)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 792, in start
    await self._default_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 799, in _default_start
    await self._actually_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 823, in _actually_start
    await child.maybe_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 851, in maybe_start
    await self.start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 792, in start
    await self._default_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 799, in _default_start
    await self._actually_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 816, in _actually_start
    await self.on_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 1126, in on_start
    await self._fut
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/cli/base.py", line 606, in execute
    await self.run(*args, **kwargs)
  File "/Users/kurtwall/code/test-faust-app/app.py", line 14, in ask_example
    resp = await example_agent.ask(event)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/agent.py", line 800, in ask
    await app._reply_consumer.add(p.correlation_id, p)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/replies.py", line 168, in add
    await self._start_fetcher(reply_topic)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/replies.py", line 177, in _start_fetcher
    await topic.maybe_declare()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/utils/futures.py", line 58, in __call__
    result = await self.fun(*self.args, **self.kwargs)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/topics.py", line 481, in maybe_declare
    await self.declare()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/topics.py", line 496, in declare
    await producer.create_topic(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1264, in create_topic
    await cast(Transport, self.transport)._create_topic(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1478, in _create_topic
    await wrap()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/utils/futures.py", line 58, in __call__
    result = await self.fun(*self.args, **self.kwargs)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1565, in _really_create_topic
    raise for_code(code)(f"Cannot create topic: {topic} ({code}): {reason}")
kafka.errors.InvalidReplicationFactorError: [Error 38] InvalidReplicationFactorError: Cannot create topic: f-reply-5d2dda89-c3d8-4831-b5b6-eb4ba7588d25 (38): Replication factor must be larger than 0.
[2022-10-04 12:59:30,468] [37191] [ERROR] [^Worker]: Error while stopping child <ask_example: stopping >: InvalidReplicationFactorError('Cannot create topic: f-reply-5d2dda89-c3d8-4831-b5b6-eb4ba7588d25 (38): Replication factor must be larger than 0.')
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 919, in _default_stop_children
    await asyncio.shield(child.stop())
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 894, in stop
    await self.on_stop()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 1137, in on_stop
    fut.result()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/worker.py", line 279, in execute_from_commandline
    self.loop.run_until_complete(self._starting_fut)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 792, in start
    await self._default_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 799, in _default_start
    await self._actually_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 823, in _actually_start
    await child.maybe_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 851, in maybe_start
    await self.start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 792, in start
    await self._default_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 799, in _default_start
    await self._actually_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 816, in _actually_start
    await self.on_start()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 1126, in on_start
    await self._fut
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/cli/base.py", line 606, in execute
    await self.run(*args, **kwargs)
  File "/Users/kurtwall/code/test-faust-app/app.py", line 14, in ask_example
    resp = await example_agent.ask(event)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/agent.py", line 800, in ask
    await app._reply_consumer.add(p.correlation_id, p)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/replies.py", line 168, in add
    await self._start_fetcher(reply_topic)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/agents/replies.py", line 177, in _start_fetcher
    await topic.maybe_declare()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/utils/futures.py", line 58, in __call__
    result = await self.fun(*self.args, **self.kwargs)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/topics.py", line 481, in maybe_declare
    await self.declare()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/topics.py", line 496, in declare
    await producer.create_topic(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1264, in create_topic
    await cast(Transport, self.transport)._create_topic(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1478, in _create_topic
    await wrap()
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/utils/futures.py", line 58, in __call__
    result = await self.fun(*self.args, **self.kwargs)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/drivers/aiokafka.py", line 1565, in _really_create_topic
    raise for_code(code)(f"Cannot create topic: {topic} ({code}): {reason}")
kafka.errors.InvalidReplicationFactorError: [Error 38] InvalidReplicationFactorError: Cannot create topic: f-reply-5d2dda89-c3d8-4831-b5b6-eb4ba7588d25 (38): Replication factor must be larger than 0.
```

It seems like there's an issue for this already #103 and that should _certainly_ be fixed. However, if you can get a usable topic, you can feed it as the `reply_to` argument:

```python
event_topic = app.topic('example-reply', internal=False)
await example_agent.ask(event, reply_to=event_topic, correlation_id='1')
```

But the `await example_agent.ask()` just hangs. I can see output in the worker, so I know the agent got it and the topic has this message `{"key":null,"value":"this is a test","correlation_id":"1","__faust":{"ns":"@ReqRepResponse"}}` but `ask()` still doesn't seem to know that's the reply.

I also tried to use the `event_topic` as a sink of the agent and also use it as the `reply_to` argument of the command, which produces this error:
```
[2022-10-04 13:34:25,455] [43899] [ERROR] [^-Consumer]: Drain messages raised: KeyError(TopicPartition(topic='example-reply', partition=0))
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/consumer.py", line 1197, in _drain_messages
    await self.wait_first(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 751, in wait_first
    f.result()  # propagate exceptions
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/conductor.py", line 274, in on_message
    return await get_callback_for_tp(message.tp)(message)
KeyError: TopicPartition(topic='example-reply', partition=0)
[2022-10-04 13:34:25,471] [43899] [ERROR] [^-Fetcher]: Crashed reason=KeyError(TopicPartition(topic='local.dredd-apps.dredd.channels.test_agent', partition=0))
Traceback (most recent call last):
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 835, in _execute_task
    await task
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/consumer.py", line 177, in _fetcher
    await consumer._drain_messages(self)
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/consumer.py", line 1197, in _drain_messages
    await self.wait_first(
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/mode/services.py", line 751, in wait_first
    f.result()  # propagate exceptions
  File "/Users/kurtwall/.pyenv/versions/3.8.5/envs/faust-example/lib/python3.8/site-packages/faust/transport/conductor.py", line 274, in on_message
    return await get_callback_for_tp(message.tp)(message)
KeyError: TopicPartition(topic='example-reply', partition=0)
```

I created a PR in https://github.com/faust-streaming/faust/pull/375 to address part of the issue and `examples/advanced/rpc.py` works fine now. When encountering the errors:
```
[2022-10-04 18:10:49,549] [1938905] [ERROR] Topic RPC__mul not found in cluster metadata 
[2022-10-04 18:10:49,550] [1938905] [ERROR] Topic RPC__pow not found in cluster metadata 
```
I manually created those topics and reran the code, everything works fine now.

--- a/faust/agents/replies.py
+++ b/faust/agents/replies.py
@@ -174,8 +174,8 @@ class ReplyConsumer(Service):
             self._fetchers[topic_name] = None
             # declare the topic
             topic = self._reply_topic(topic_name)
-            self.app.topics.add(topic)
             await topic.maybe_declare()
+            self.app.topics.add(topic)
             await self.sleep(3.0)
             # then create the future
             self._fetchers[topic_name] = self.add_future(self._drain_replies(topic))

--- a/faust/agents/replies.py
+++ b/faust/agents/replies.py
@@ -174,6 +174,7 @@ class ReplyConsumer(Service):
             self._fetchers[topic_name] = None
             # declare the topic
             topic = self._reply_topic(topic_name)
+            self.app.topics.add(topic)
             await topic.maybe_declare()
             await self.sleep(3.0)
             # then create the future

--- a/faust/transport/conductor.py
+++ b/faust/transport/conductor.py
@@ -338,10 +338,6 @@ class Conductor(ConductorT, Service):
                     self._acking_topics.add(topic)
                 self._topic_name_index[topic].add(channel)
 
-        if self.app.client_only:
-            self._init_tp_index()
-            self._update_callback_map()
-
         return self._topic_name_index
 
     async def on_partitions_assigned(self, assigned: Set[TP]) -> None:
@@ -352,6 +348,7 @@ class Conductor(ConductorT, Service):
         T(self._update_callback_map)()
 
     async def on_client_only_start(self) -> None:
+        self.log.info("Starting client only mode with %s topics...", self._topics)
         self._init_tp_index()
         self._update_callback_map()
 
@@ -420,7 +417,7 @@ class Conductor(ConductorT, Service):
 
     def _topic_contain_unsubscribed_topics(self, topic: TopicT) -> bool:
         index = self._topic_name_index
-        return bool(index and any(t not in index for t in topic.topics))
+        return bool(any(t not in index for t in topic.topics))
 
     def discard(self, topic: Any) -> None:
         """Unregister topic from conductor."""

--- a/faust/transport/conductor.py
+++ b/faust/transport/conductor.py
@@ -348,15 +348,12 @@ class Conductor(ConductorT, Service):
         T(self._update_callback_map)()
 
     async def on_client_only_start(self) -> None:
-        self._init_tp_index()
-        self._update_callback_map()
-
-    def _init_tp_index(self) -> None:
         tp_index = self._tp_index
         for topic in self._topics:
             for subtopic in topic.topics:
                 tp = TP(topic=subtopic, partition=0)
                 tp_index[tp].add(topic)
+        self._update_callback_map()
 
     def _update_tp_index(self, assigned: Set[TP]) -> None:
         assignmap = tp_set_to_map(assigned)

--- a/faust/transport/conductor.py
+++ b/faust/transport/conductor.py
@@ -338,8 +338,10 @@ class Conductor(ConductorT, Service):
                     self._acking_topics.add(topic)
                 self._topic_name_index[topic].add(channel)
 
-        self._init_tp_index()
-        self._update_callback_map()
+        if self.app.client_only:
+            self._init_tp_index()
+            self._update_callback_map()
+
         return self._topic_name_index
 
     async def on_partitions_assigned(self, assigned: Set[TP]) -> None:

--- a/faust/transport/conductor.py
+++ b/faust/transport/conductor.py
@@ -338,6 +338,8 @@ class Conductor(ConductorT, Service):
                     self._acking_topics.add(topic)
                 self._topic_name_index[topic].add(channel)
 
+        self._init_tp_index()
+        self._update_callback_map()
         return self._topic_name_index
 
     async def on_partitions_assigned(self, assigned: Set[TP]) -> None:
@@ -348,12 +350,15 @@ class Conductor(ConductorT, Service):
         T(self._update_callback_map)()
 
     async def on_client_only_start(self) -> None:
+        self._init_tp_index()
+        self._update_callback_map()
+
+    def _init_tp_index(self) -> None:
         tp_index = self._tp_index
         for topic in self._topics:
             for subtopic in topic.topics:
                 tp = TP(topic=subtopic, partition=0)
                 tp_index[tp].add(topic)
-        self._update_callback_map()
 
     def _update_tp_index(self, assigned: Set[TP]) -> None:
         assignmap = tp_set_to_map(assigned)

--- a/faust/transport/conductor.py
+++ b/faust/transport/conductor.py
@@ -348,7 +348,6 @@ class Conductor(ConductorT, Service):
         T(self._update_callback_map)()
 
     async def on_client_only_start(self) -> None:
-        self.log.info("Starting client only mode with %s topics...", self._topics)
         self._init_tp_index()
         self._update_callback_map()
 

--- a/tests/unit/agents/test_replies.py
+++ b/tests/unit/agents/test_replies.py
@@ -1,6 +1,6 @@
 import asyncio
 import json
-from unittest.mock import Mock
+from unittest.mock import Mock, MagicMock
 
 import pytest
 
@@ -192,7 +192,7 @@ class Test_ReplyConsumer:
     async def test_start_fetcher(self, *, c):
         c._drain_replies = Mock()
         c._reply_topic = Mock(
-            return_value=Mock(
+            return_value=MagicMock(
                 maybe_declare=AsyncMock(),
             ),
         )

--- a/tests/unit/agents/test_replies.py
+++ b/tests/unit/agents/test_replies.py
@@ -1,6 +1,6 @@
 import asyncio
 import json
-from unittest.mock import Mock, MagicMock
+from unittest.mock import MagicMock, Mock
 
 import pytest
 

